src/base/memory_monitor_sr.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * memory_monitor_sr.cpp - implementation of memory monitoring manager
+ */
+
+#include <stdexcept>
+#include <cstring>
+
+#include "perf.hpp"
+#include "error_manager.h"
+#include "memory_monitor_sr.hpp"
+
+#if defined (SERVER_MODE)
+#include "log_impl.h"
+#endif
+
+
+#ifdef SERVER_MODE
+namespace cubperf
+{
+  memory_monitoring_manager *mmm_Gl = NULL;
+
+  int mmm_component::add_subcomponent (const char *name)
+  {
+    return 0;
+  }
+
+  mmm_module::mmm_module (const char *name, MMM_COMP_INFO *info)
+  {
+    m_module_name = new char[strlen (name) + 1];
+    strcpy (m_module_name, name);
+
+    /* register component and subcomponent information
+     * add component or subcomponent */
+    int cnt = 0;
+    while (info[cnt].idx != MMM_STAT_LAST)
+      {
+	bool comp_skip = false;
+	bool subcomp_skip = false;
+	int comp_idx = this->m_max_idx, subcomp_idx = this->m_max_idx;
+	int i;
+	if (info[cnt].comp_name.size())
+	  {
+	    for (i = 0; i < m_component.size(); i++)
```

Comment:
for-range is better unless the index is needed.
```suggestion
	    for  (const auto& comp :  m_component)
```

Reply:
I believe using a regular for loop is better in _mmm_module::mmm_module_ because we need to extract the index of component and subcomponent for _m_comp_idx_map_.

Reply:
Oh, my bad. Forget it :(